File: wooseongshin/2월/7일/h-index.java
import java.util.Arrays;

public class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);

        for (int h = 0; h <= citations[citations.length - 1]; h++) {
            int below = belowCount(citations, h);
            int above = aboveCount(citations, h);
            if (below <= h && above >= h && answer < h) {
                answer = h;
            }
        }

        return answer;
    }

    private int belowCount(int[] citations, int h) {
        int count = 0;
        for (int citation : citations) {
            if (citation <= h) {
                count++;
            } else {
                break;
            }
        }
        return count;
    }

    private int aboveCount(int[] citations, int h) {
        int count = 0;
        for (int citation : citations) {
            if (citation >= h) {
                count++;
            }
        }
        return count;
    }
}

File: Eunleelee/2024-02/2월 8일/H-Index.py
def solution(citations):
    answer = []
    citations.sort() 
    n = len(citations)
    
    for i in range(citations[-1]+1):
        for j, citation in enumerate(citations):
            if i <= citation:
                h_higher_citation = n - j
                break
        if h_higher_citation >= i:
            answer.append(i)
        
    return answer[-1]